client/state/sites/reducer.js

```diff
@@ -19,12 +25,21 @@ import { SITE_RECEIVE } from 'state/action-types';
 export function items( state = {}, action ) {
 	switch ( action.type ) {
 		case SITE_RECEIVE:
-			state = Object.assign( {}, state, {
+			return Object.assign( {}, state, {
 				[ action.site.ID ]: action.site
 			} );
-			break;
+		case SERIALIZE:
+			// scrub _events, _maxListeners, and other misc functions
+			const sites = Object.keys( state ).map( ( siteID ) => {
+				let plainJSObject = Object.assign( {}, state[ siteID ] );
+				plainJSObject = pick( plainJSObject, ( value ) => ! isFunction( value ) );
```

Comment:
Potential optimization by avoiding the `Object.assign` altogether, since `pick` does not mutate the original object:

```js
let plainJSObject = pick( state[ siteID ], ( value ) => ! isFunction( value ) );
```

I'd much rather we fix our current usage of sites to not store these properties in the first place, but seems like we're far enough along with this pull request that it's not worth holding up.